lib/check_params.h

```diff
@@ -1026,14 +1026,32 @@ void printQudaGaugeObservableParam(QudaGaugeObservableParam *param)
 #ifdef INIT_PARAM
   P(su_project, QUDA_BOOLEAN_FALSE);
   P(compute_plaquette, QUDA_BOOLEAN_FALSE);
+  P(compute_polyakov_loop, QUDA_BOOLEAN_FALSE);
+  P(compute_gauge_loop_trace, QUDA_BOOLEAN_FALSE);
+  P(traces, nullptr);
+  P(input_path_buff, nullptr);
+  P(path_length, nullptr);
+  P(loop_coeff, nullptr);
+  P(num_paths, 0);
```

Comment:
`num_paths`, `max_length` and `factor` should be invalid initialized here.  I think what should be done here is that lines 1047-1051 should just be moved below the `endif` on line 1055, and lines 1035-1037 should be deleted.

Reply:
Makes sense, changed locally, will push after some cursory testing

Reply:
So in practice it couldn't be moved out of the `if`/`endif`/etc outright, due to different definitions of `P` and the local struct being `*param` or `ret`, but I think I found a happy medium.

Reply:
https://github.com/lattice/quda/commit/0eacdf5a4fa5ffd1631db7509707665b0b59f8b4